src/test/java/guitests/PersonEditGuiTest.java

```diff
@@ -42,35 +45,118 @@ public void editPerson() {
+        editPersonDialog.clickOk();
+
+        assertTrue(editPersonDialog.isInputValidationErrorDialogShown());
+        editPersonDialog.dissmissErrorMessage("Invalid Fields");
+        assertFalse(editPersonDialog.isInputValidationErrorDialogShown());
+    }
+
+    @Test
+    public void cancelOperation_usingAccelerator() {
+
+        //Delete
+        PersonCardHandle aliceCard = personListPanel.selectCard(td.alice);
+        personListPanel.use_PERSON_DELETE_ACCELERATOR();
+        assertTrue(aliceCard.isShowingGracePeriod("Deleting"));
+
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertFalse(aliceCard.isShowingGracePeriod("Deleting"));
+        assertEquals(statusBar.getText(), "Delete Person [ " + aliceCard.getFirstName() + " "
+                     + aliceCard.getLastName() + " ] was cancelled.");
+
+        //Edit
+        Person newAlice = new PersonBuilder(td.alice.copy()).withFirstName("Alicia").withLastName("Brownstone")
+                .withStreet("Updated street").withCity("Singapore").withPostalCode("123123")
+                .withBirthday("01.01.1979").withGithubUsername("alicebrown123").withTags(td.colleagues, td.friends).build();
+
+        //Get a reference to the card displaying Alice's details
+        PersonCardHandle alicePersonCard = personListPanel.selectCard(td.alice);
+
+        //Edit Alice to change to new values
+        EditPersonDialogHandle editPersonDialog = personListPanel.use_PERSON_EDIT_ACCELERATOR();
+        editPersonDialog.enterNewValues(newAlice);
+        editPersonDialog.pressEnter();
+
+        mainGui.focusOnMainApp();
+        personListPanel.clickOnPerson(newAlice);
+
+        assertNotEquals(alicePersonCard, td.alice);
+        assertEquals(alicePersonCard, newAlice);
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertEquals(alicePersonCard, td.alice);
+        assertEquals(statusBar.getText(), "Edit Person [ Alice Brown -> Alicia Brownstone ] was cancelled.");
+
+        //New
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+        Person pandaWong = new PersonBuilder("Panda", "Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
+        addPersonDialog.enterNewValues(pandaWong);
+        addPersonDialog.clickOk();
+
+        mainGui.focusOnMainApp();
+        personListPanel.clickOnListView();
+        personListPanel.use_LIST_GOTO_BOTTOM_SEQUENCE();
+        personListPanel.clickOnPerson(pandaWong);
+        PersonCardHandle pandaWongCardHandle = personListPanel.getSelectedCards().get(0);
+        assertNotNull(pandaWongCardHandle);
+        assertEquals(pandaWongCardHandle, pandaWong);
+        personListPanel.use_PERSON_CHANGE_CANCEL_ACCELERATOR();
+        assertNull(personListPanel.getPersonCardHandle(pandaWong));
+    }
```

Comment:
Do you have a test case to confirm Ctrl+Z does not have any effect after the grace period is over? If not, add a todo or add such a test.